lib/ansible/modules/net_tools/haproxy.py

```diff
@@ -251,12 +252,12 @@ def execute(self, cmd, timeout=200, capture_output=True):
         """
         self.client = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
         self.client.connect(self.socket)
-        self.client.sendall('%s\n' % cmd)
+        self.client.sendall(to_bytes('%s\n' % cmd))
         result = ''
         buf = ''
         buf = self.client.recv(RECV_SIZE)
         while buf:
-            result += buf
+            result += to_text(buf)
```

Comment:
Are we guaranteed to only receive ascii text via self.client.recv? If not, then this could fail if we get the first half of a multi-byte character in buf and the second half is coming in the next chunk.  Encoding and decoding is also faster when you do it a single time on a larger buffer rather than multiple times on smaller buffers.  Between those two thoughts, perhaps it would be better to code this like:
``` python
        self.client.sendall(to_bytes('%s\n' % cmd))
        result = b''
        buf = b''
        buf = self.client.recv(RECV_SIZE)
        while buf:
            result += buf
            buf = self.client.recv(RECV_SIZE) 
        result = to_text(result, errors='surrogate_or_strict')

        if capture_output:
[...]
```